src/eko/io/manipulate.py

```diff
@@ -18,10 +19,8 @@
 SIMGRID_ROTATION = "ij,ajbk,kl->aibl"
 """Simultaneous grid rotation contraction indices."""
 
-Basis = Union[XGrid, npt.NDArray]
 
-
-def rotation(new: Optional[Basis], old: Basis, check: Callable, compute: Callable):
+def rotation(new: Optional[XGrid], old: XGrid, check: Callable, compute: Callable):
```

Comment:
If I understand correctly this function can be simplified a bit more. Since `newtarget` and `newinput` are not used anymore we can return only the second element. Also the `Optional` for `new` can be dropped.

Reply:
- returning one element is done in https://github.com/NNPDF/eko/pull/292/commits/4699064f9e4c5e368e148f1afb9614e0e9908689
- `Optional` we still need, since we're calling the function always on both sides, but one might me `None`